src/Products/Viewer/Controllers/ViewerApiController.cs

```diff
@@ -383,196 +411,318 @@ public HttpResponseMessage PrintPdf(PostedDataEntity loadDocumentRequest)
                 response.Content.Headers.ContentDisposition.FileName = Path.GetFileName(fileName);
                 return response;
             }
-            catch (System.Exception ex)
+            catch (Exception ex)
             {
                 // set exception message
                 return Request.CreateResponse(HttpStatusCode.InternalServerError, new Resources().GenerateException(ex));
             }
         }
 
+        private void SaveChangedAngleInCache(ViewerSettings settings, int pageNumber, int newAngle)
+        {
+            var cacheFolder = ((FileCache)settings.Cache).CachePath;
+            var pagesInfoPath = Path.Combine(cacheFolder, "PagesInfo.xml");
+
+            if (File.Exists(pagesInfoPath)) 
+            {
+                XDocument xdoc = XDocument.Load(pagesInfoPath);
+                var pageData = xdoc.Descendants()?.Elements("Number")?.Where(x => int.Parse(x.Value) == pageNumber)?.Ancestors("PageData");
+                var angle = pageData.Elements("Angle").FirstOrDefault();
+
+                if (angle != null)
+                {
+                    angle.Value = newAngle.ToString(CultureInfo.InvariantCulture);
+                }
+
+                xdoc.Save(pagesInfoPath);
+            }
+        }
+
+        private int GetNewAngleValue(int currentAngle, int postedAngle)
+        {
+            switch (currentAngle) 
+            {
+                case 0:
+                    return postedAngle == 90 ? 90 : 270;
+                case 90:
+                    return postedAngle == 90 ? 180 : 0;
+                case 180:
+                    return postedAngle == 90 ? 270 : 90;
+                case 270:
+                    return postedAngle == 90 ? 0 : 180;
+                default:
+                    return 0;
+            }
+        }
+
+        private Rotation GetRotationByAngle(int newAngle)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make 'GetRotationByAngle' a static method.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5242614)